app/controllers/course/material/folders_controller.rb

```diff
@@ -0,0 +1,52 @@
+class Course::Material::FoldersController < Course::ComponentController
+  load_and_authorize_resource :material_folder, through: :course, parent: false,
```

Comment:
Do you also want to rename the `material_folder` association in course ?  I'm using some names same as in v1.